/*
 * This file was automatically generated by EvoSuite
 * Sat May 09 18:18:16 GMT 2020
 */

package ElevatorSystem;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import ElevatorSystem.Elevator;
import ElevatorSystem.Environment;
import ElevatorSystem.EvilPerson;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class EvilPerson_ESTest extends EvilPerson_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      int[] intArray0 = new int[3];
      Environment environment0 = new Environment(762);
      EvilPerson evilPerson0 = new EvilPerson("", 0, 0, 762, intArray0, environment0);
      // Undeclared exception!
      try { 
        evilPerson0.enterElevator((Elevator) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.Person", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      int[] intArray0 = new int[8];
      Environment environment0 = new Environment(2665);
      EvilPerson evilPerson0 = new EvilPerson("", 0, 0, 2665, intArray0, environment0);
      Elevator elevator0 = new Elevator(environment0, true);
      // Undeclared exception!
      try { 
        evilPerson0.enterElevator(elevator0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 2665
         //
         verifyException("ElevatorSystem.Elevator", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      int[] intArray0 = new int[8];
      EvilPerson evilPerson0 = null;
      try {
        evilPerson0 = new EvilPerson("down", 0, (-1), 0, intArray0, (Environment) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.Person", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      int[] intArray0 = new int[10];
      Environment environment0 = new Environment(2714);
      EvilPerson evilPerson0 = new EvilPerson(")", 10, 10, 10, intArray0, environment0);
      Elevator elevator0 = new Elevator(environment0, true, 10, true);
      evilPerson0.enterElevator(elevator0);
      assertTrue(elevator0.areDoorsOpen());
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      int[] intArray0 = new int[8];
      Environment environment0 = new Environment(0);
      EvilPerson evilPerson0 = null;
      try {
        evilPerson0 = new EvilPerson("6>ign", (-597), (-2881), 440, intArray0, environment0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // -2881
         //
         verifyException("ElevatorSystem.Environment", e);
      }
  }
}
